### PR TITLE
gofmt / golint / minor code changes

### PR DESCRIPTION
**Pull request by virtuallynathan**
* A number of formatting / comment changes.
  * Following the [Go Code Review Comments](https://github.com/golang/go/wiki/CodeReviewComments) guidelines and [Effective Go on Commentary](https://golang.org/doc/effective_go.html#commentary) where possible (for exported functions and structs)
* removes verbose declarations
* remove unused range variable
* remove runtime.GOMAXPROCS, this was [set to NumCPUs in Go1.5](https://docs.google.com/document/d/1At2Ls5_fhJQ59kDK2DFVhFu3g5mATSXqqV5QrxinasI/edit)
* Currently the code (at least on OS X) will panic if not run as root. Added a comment to the error output before the panic. I'll perhaps submit a later pull request to handle this.

I have signed the CLA

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -216,14 +206,12 @@ func TcpReceiver(done <-chan struct{}, af string, targetAddr string, probePortSt
 	return out, nil
 }
 
-//
-// This runs on its own collecting Icmp responses until its explicitly told to stop
-//
-func IcmpReceiver(done <-chan struct{}, af string) (chan interface{}, error) {
+// ICMPReceiver runs on its own collecting Icmp responses until its explicitly told to stop
```
Comment (insomniacslk) [inline main.go]: Can you change "Icmp" to "ICMP" here?

Comment (virtuallynathan) [inline main.go]: It should be changed there, I just compared to my banch

Comment (insomniacslk) [inline main.go]: I don't get what you mean.

To clarify: I'm asking if you can amend your diff to make the word "Icmp" uppercase in that comment, for consistency with the rest

Comment (virtuallynathan) [inline main.go]: Oh, yep, totally did miss the comment. I was looking at the wrong line.

**Pull request merged**